### PR TITLE
Closes #3247: QrScanner: Remove image cropping before processing

### LINKED ISSUE #3247: QrScanner: Incorrect image cropping before processing
This is the cause of: https://github.com/mozilla-mobile/fenix/issues/2565

The problem is that we're incorrectly cropping the image before processing.

### PR DESCRIPTION
**Pull request by csadilek**
This fixes: https://github.com/mozilla-mobile/fenix/issues/2565 (due to incorrect image cropping)

The scanner works a lot better in general and is still fast even on old devices.

### CODE REVIEW

**Comment by csadilek** [comment]
I am gonna try and write a test for this.

**Comment by csadilek** [comment]
Test added.

File: components/feature/qr/src/test/java/mozilla/components/feature/qr/QrFragmentTest.kt [hunk 1]
```diff
@@ -127,6 +129,21 @@ class QrFragmentTest {
...
+        val reader = mock(MultiFormatReader::class.java)
+        val task = QrFragment.AsyncScanningTask(listener, reader)
+        val imageCaptor = argumentCaptor<BinaryBitmap>()
+
+        val bitmap = mock(BinaryBitmap::class.java)
+
+        QrFragment.qrState = QrFragment.STATE_DECODE_PROGRESS
+        task.processImage(bitmap)
+        verify(reader).decodeWithState(imageCaptor.capture())
+        assertSame(bitmap, imageCaptor.value)
```
Comment (vladikoff) [inline components/feature/qr/src/test/java/mozilla/components/feature/qr/QrFragmentTest.kt]: nice 👍

**Comment by vladikoff** [comment]
` :service-glean` task failed 😢

**Comment by vesta0** [comment]
Let's keep the reticle please.

**Pull request merged**